### PR TITLE
[Docs] Update Dev. Doc. on how to add a new relay operator

### PR DESCRIPTION
**Pull request by AndrewZhaoLuo**
Updates the tutorial and goes more in depth by using a more substantial operator than simple element-wise addition.

Instead we follow how PR #7334 and PR #7722 added cumulative product and addition operators.

Testing:
Ran
`restview docs/dev/relay_add_op.rst`

And read the document.

### CODE REVIEW

**Comment by AndrewZhaoLuo** [comment]
@masahi PTAL. Also Chris Hoge.

File: docs/dev/relay_add_op.rst [hunk 1]
```diff
@@ -15,75 +15,297 @@
...
+        .describe(
+            R"doc(Return the cumulative product of the elements along a given axis.)doc" TVM_ADD_FILELINE)
+        .set_num_inputs(1)
+        .add_argument("data", "Tensor", "The input tensor.")
+        .set_support_level(3)
+        .add_type_rel("Cumprod", ScanopRel)
+        .set_attr<TOpPattern>("TOpPattern", kOpaque);
+
+In this case the ``TOpPattern`` is a hint to the compiler on the pattern of computation the operator does, which might be
+useful for reordering loops and fusing operators. ``kOpaque`` tells TVM not to not bother trying to fuse this operator.
```
Comment (masahi) [inline docs/dev/relay_add_op.rst:185]: Let's drop `reordering loops`, `TOpPattern` is only used for operater fusion. 

A typo: `not to not bother`

File: docs/dev/relay_add_op.rst [hunk 2]
```diff
@@ -15,75 +15,297 @@
...
+
+While we've now defined the interface for the operation but still have not 
+told TVM how to perform the actual calculations for cumulative sum and product. 
+
+Writing this code is outside the scope of the tutorial. For now, we assume
+we have a well tested implementation for the operation's compute. For 
+more details on how to do this, we recommend looking up the tutorials
+on `tensor expressions`_, `TVM's operator inventory (topi)`_ and looking at the 
+examples cumulative sum and product found in `python/tvm/topi/scan.py`_ and 
+`python/tvm/topi/cuda/scan.py`_. In the case of our cumulative sum and product operations
```
Comment (masahi) [inline docs/dev/relay_add_op.rst:198]: Multiple references to the same file

File: docs/dev/relay_add_op.rst [hunk 3]
```diff
@@ -15,75 +15,297 @@
...
+        .set_attr<TOpPattern>("TOpPattern", kOpaque);
+
+In this case the ``TOpPattern`` is a hint to the compiler on the pattern of computation the operator does, which might be
+useful for reordering loops and fusing operators. ``kOpaque`` tells TVM not to not bother trying to fuse this operator. 
+
+4. Defining the Compute of the Operation
+----------------------------------------
+
+While we've now defined the interface for the operation but still have not 
+told TVM how to perform the actual calculations for cumulative sum and product.
```
Comment (masahi) [inline docs/dev/relay_add_op.rst:191]: Need to polish this sentence

File: docs/dev/relay_add_op.rst [hunk 4]
```diff
@@ -15,75 +15,297 @@
...
+.. _tensor expressions: https://tvm.apache.org/docs/tutorials/get_started/tensor_expr_get_started.html
+.. _TVM's operator inventory (topi): https://tvm.apache.org/docs/tutorials/topi/intro_topi.html
+.. _TIR: https://tvm.apache.org/docs/dev/index.html?highlight=tir#tvm-tir
+.. _python/tvm/topi/scan.py: https://github.com/apache/tvm/blob/main/python/tvm/topi/scan.py
+.. _python/tvm/topi/cuda/scan.py: https://github.com/apache/tvm/blob/main/python/tvm/topi/cuda/scan.py
+
+5. Hooking up Compute and Strategy with Relay
+---------------------------------------------
+
+After you have implemented how your function can be computed we now need to glue it to our
```
Comment (masahi) [inline docs/dev/relay_add_op.rst:211]: After you have implemented your compute function

**Comment by masahi** [comment]
@hogepodge

File: docs/dev/relay_add_op.rst (hunk 1 above)
Comment (AndrewZhaoLuo) [reply docs/dev/relay_add_op.rst]: Done

File: docs/dev/relay_add_op.rst (hunk 2 above)
Comment (AndrewZhaoLuo) [reply docs/dev/relay_add_op.rst]: They actually are different files. One is the generic implementation and one is the cuda/gpu implementation. I've made this clearer in the text

File: docs/dev/relay_add_op.rst (hunk 3 above)
Comment (AndrewZhaoLuo) [reply docs/dev/relay_add_op.rst]: PTAL

File: docs/dev/relay_add_op.rst (hunk 4 above)
Comment (AndrewZhaoLuo) [reply docs/dev/relay_add_op.rst]: Done

**Comment by AndrewZhaoLuo** [comment]
@masahi Thanks for the comments. PTAL

**Review by masahi — APPROVED** [review]

**Comment by masahi** [comment]
Thanks @AndrewZhaoLuo

**Pull request merged**